### PR TITLE
SNOW-1573367 Fix not raise NotImplementedError for seting cell with list

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1573367 Fix not raise NotImplementedError for seting cell with list

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

We failed to raise the error for some cases. This PR fixed it.

### CODE REVIEW

File: tests/integ/modin/series/test_at.py [hunk 1]
```diff
@@ -129,3 +129,12 @@ def test_at_neg(
 ):
     with pytest.raises(KeyError):
         default_index_snowpark_pandas_series.at[key]
+
+
+@sql_count_checker(query_count=0)
+def test_raise_set_cell_with_list_like_value_error():
```
Comment (sfc-gh-helmeleegy) [inline tests/integ/modin/series/test_at.py:135]: Nit: Any particular reason this test is added under the `series` directory, while the three others under the `frame` directory?

I'd suggest either add all of them under one directory (say `frame`) or perhaps better to keep the dataframe parts of the tests under `frame` and the series parts under `series`.

**Review by sfc-gh-helmeleegy — COMMENTED** [review]
Looks good over all - except for a small nit.

**Review by sfc-gh-helmeleegy — APPROVED** [review]

**Review by sfc-gh-evandenberg — APPROVED** [review]
LGTM

File: tests/integ/modin/series/test_at.py (hunk 1 above)
Comment (sfc-gh-azhan) [reply tests/integ/modin/series/test_at.py:135]: oh, I should put under `frame`. Good catch!

**Pull request merged**